### ISSUE TITLE
[Bug]: Token validity check should be synchronous

### ISSUE THREAD

**Issue by vedantmgoyal2009** [russellbanks/Komac#36]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Whenever I try to run `new` or `update` commands, token validity checks obstruct the package identifier prompt, which is bad in terms of UX.

Instead, of checking in the background, we can display some kind of animation like a loading circle, and after validation checks are complete, we can ask for a package identifier, or token if the current token is invalid.

![image](https://user-images.githubusercontent.com/83997633/214764611-8877eab4-e0c9-4673-9106-bababdca5fcd.png)

**Comment by russellbanks** [comment]
I ideally want to avoid synchronisity in Komac because having to wait for something to complete doesn't feel like a great user experience. Yamlcreate has this for a second or so as it retrieves all the schemas when it's first ran, so I want to avoid having anything blocking that does the same.

For token validity checks, I've adjusted it in https://github.com/russellbanks/Komac/commit/a197424ad21f2654ded601f381b32858377a7c2f:

There are now 2 "breakpoints" during the Package Identifier prompt. [The first one](https://github.com/russellbanks/Komac/blob/a197424ad21f2654ded601f381b32858377a7c2f/src/main/kotlin/data/shared/PackageIdentifier.kt#L31) checks whether there is a currently stored token, as this can be retrieved fairly quickly from Windows Credentials. If there isn't the user is prompted for it before the Package Identifier prompt. [The second one](https://github.com/russellbanks/Komac/blob/a197424ad21f2654ded601f381b32858377a7c2f/src/main/kotlin/data/shared/PackageIdentifier.kt#L53) is after the Package Identifier prompt and checks whether it's valid. If it's not the user will be prompted for a new token.

This is because retrieving the token takes a very small amount of time whilst checking if the token is valid requires a request to GitHub, taking a second or longer. The way I have changed it means that it can do the longer validity check in the background while the user is being asked for the Package Identifier. In a situation where we have the token stored and it's valid, the user shouldn't notice that there are any checks happening.
